plugins/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureBlobContainerRetriesTests.java

```diff
@@ -198,6 +214,57 @@ public void testReadBlobWithRetries() throws Exception {
         }
     }
 
+    public void testReadRangeBlobWithRetries() throws Exception {
+        final int maxRetries = randomIntBetween(1, 5);
+        final CountDown countDownHead = new CountDown(maxRetries);
+        final CountDown countDownGet = new CountDown(maxRetries);
+        final byte[] bytes = randomBlobContent();
+        httpServer.createContext("/container/read_range_blob_max_retries", exchange -> {
+            Streams.readFully(exchange.getRequestBody());
+            if ("HEAD".equals(exchange.getRequestMethod())) {
+                if (countDownHead.countDown()) {
+                    exchange.getResponseHeaders().add("Content-Type", "application/octet-stream");
+                    exchange.getResponseHeaders().add("x-ms-blob-content-length", String.valueOf(bytes.length));
+                    exchange.getResponseHeaders().add("x-ms-blob-type", "blockblob");
+                    exchange.sendResponseHeaders(RestStatus.OK.getStatus(), -1);
+                    exchange.close();
```

Comment:
Maybe `try { ... } finally { exchange.close()}` around the whole handler so we don't get weird exceptions from the REST mock if an assertion trips?

Reply:
Sure